src/kw_manager.sh

```diff
@@ -265,15 +282,245 @@ function rename_kw_manager_group()
   fi
 }
 
+# This function add a new contact to a given kw mail group
+#
+# @contacts_string: The string with all the contacts informations
+# @group_name: The name of the group which the contacts will be added.
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function add_kw_manager_contacts()
+{
+  local contacts_list="$1"
+  local group_name="$2"
+  local group_id
+  declare -A _contacts_array
+  declare -A _new_contacts_array
+
+  if [[ -z "$contacts_list" ]]; then
+    complain 'The contacts list is empty'
+    return 22 #EINVAL
+  fi
+
+  check_existent_group "$group_name" && exit_msg 'Error while checking group existence'
+
+  group_id="$?"
+
+  if [[ $(split_contact_infos "$contacts_list" '_contacts_array') ]]; then
+    complain 'Error while parsing contacts list'
+  fi
+
+  if [[ $(add_contacts '_contacts_array') ]]; then
+    complain 'Error while adding new contacts to the database'
+  fi
+
+  if [[ $(add_contact_group '_contacts_array' "$group_id") ]]; then
+    complain 'Error while associating contact to the group'
+  fi
+
+  return 0
+}
+
+# This function split the columns in the given contact string passed
+# as parameter and create an associative array with the infos.
+#
+# @contacts_list: An string formed as "CONTACT_NAME <CONTACT_EMAIL>, CONTACT_NAME <CONTACT_EMAIL>, ..."
+# @contacts_array: An associative array formed as "<CONTACT_EMAIL>:<CONTACT_NAME>"
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function split_contact_infos()
```

Comment:
I feel this could be a generic function that other parts of kw can use, such as the email and signature code.